application/src/main/java/org/togetherjava/tjbot/commands/moderation/scam/ScamBlocker.java

```diff
@@ -0,0 +1,356 @@
+    }
+
+    private void takeActionAutoDeleteAndQuarantine(@NotNull GuildMessageReceivedEvent event) {
+        addScamToHistory(event);
+        logScamMessage(event);
+        deleteMessage(event);
+        quarantineAuthor(event);
+        dmUser(event);
+        reportScamMessage(event, "Detected and handled scam", null);
+    }
+
+    private void addScamToHistory(@NotNull GuildMessageReceivedEvent event) {
+        scamHistoryStore.addScam(event.getMessage(), MODES_WITH_IMMEDIATE_DELETION.contains(mode));
+    }
+
+    private void logScamMessage(@NotNull GuildMessageReceivedEvent event) {
+        logger.warn("Detected a scam message ('{}') from user '{}' in channel '{}' of guild '{}'.",
+                event.getMessageId(), event.getAuthor().getId(), event.getChannel().getId(),
+                event.getGuild().getId());
+    }
+
+    private void deleteMessage(@NotNull GuildMessageReceivedEvent event) {
+        event.getMessage().delete().queue();
+    }
+
+    private void quarantineAuthor(@NotNull GuildMessageReceivedEvent event) {
+        quarantineAuthor(event.getGuild(), event.getMember(), event.getJDA().getSelfUser());
+    }
+
+    private void quarantineAuthor(@NotNull Guild guild, @NotNull Member author,
+            @NotNull SelfUser bot) {
+        String reason = "User posted scam that was automatically detected";
+
+        actionsStore.addAction(guild.getIdLong(), bot.getIdLong(), author.getIdLong(),
+                ModerationAction.QUARANTINE, null, reason);
+
+        guild
+            .addRoleToMember(author,
+                    ModerationUtils.getQuarantinedRole(guild, config).orElseThrow())
+            .reason(reason)
+            .queue();
+    }
+
+    private void reportScamMessage(@NotNull GuildMessageReceivedEvent event,
+            @NotNull String reportTitle, @Nullable ActionRow confirmDialog) {
+        Guild guild = event.getGuild();
+        Optional<TextChannel> reportChannel = getReportChannel(guild);
+        if (reportChannel.isEmpty()) {
+            logger.warn(
+                    "Unable to report a scam message, did not find a report channel matching the configured pattern '{}' for guild '{}'",
+                    reportChannelPattern, guild.getName());
+            return;
+        }
+
+        User author = event.getAuthor();
+        MessageEmbed embed =
+                new EmbedBuilder().setDescription(event.getMessage().getContentStripped())
+                    .setTitle(reportTitle)
+                    .setAuthor(author.getAsTag(), null, author.getAvatarUrl())
+                    .setTimestamp(event.getMessage().getTimeCreated())
+                    .setColor(AMBIENT_COLOR)
+                    .setFooter(author.getId())
+                    .build();
+        Message message =
+                new MessageBuilder().setEmbeds(embed).setActionRows(confirmDialog).build();
+
+        reportChannel.orElseThrow().sendMessage(message).queue();
```

Comment:
```suggestion
        reportChannel.orElseThrow().sendMessage(message)
            .setEmbeds(embed)
            .setActionRows(confirmDialog)
            .queue();
```

This @Zabuzard

Reply:
its not possible. we have a special situation here.

i want to send a message with no message content, but emebeds and action rows.

problem is, `sendMessage("")` and `sendMessage(null)` both dont work. and `sendEmbeds(...)` doesnt take action rows (does it?)

Reply:
`sendMessageEmbeds(embed)`

Reply:
fair 😆

Reply:
Actually. There is an issue. This `setActionRows` only allows a non-nullable action row (or non-empty action row lists).

The other one allows a nullable row (or empty lists).

So if we still want to stick with that design, I have to create an if and break the fluent-chain:
```java
var foo = reportChannel.orElseThrow().sendMessageEmbeds(embed);
if (confirmDialog != null) {
  foo = foo.setActionRows(confirmDialog);
}
foo.queue();
```
at which point I prefer the previous solution.

Reply:
In that case nvm since that's way too hard to reaad.